### PR TITLE
update HowToDebug.md

### PR DESCRIPTION
**Pull request by Crysple**

### CODE REVIEW

File: docs/HowToDebug.md [hunk 1]
```diff
@@ -1,4 +1,50 @@
...
+
+Usually in webUI, you can click `+` in the left of every trial to expand it to see each trial's log path. Also there is another directory under experiment root directory, named `trials-local`, which stores all the trials run. Every trial has a unique id that is the name of its root directory. In this directory, a file named `stderr` records trial error and another named `trial.log` records this trial's log. 
+
+## Different kinds of errors
+
+There are different kinds of errors. However, they can be divided into three categories based on their severity. So when nni fails, check each part sequentially.
+
+### **nnimanger** Failed
+
+Usually this is the most serious error. When this happens, the whole experiment fails and no trial will be run.
```
Comment (xuehui1991) [inline docs/HowToDebug.md]: could you post some common error example here?

Comment (Crysple) [reply docs/HowToDebug.md]: Sure, will try to add some.

File: docs/HowToDebug.md [hunk 2]
```diff
@@ -1,4 +1,80 @@
 **How to Debug in NNI**
 ===
 
-*Coming soon*
+## Overview
+
+There are three parts that might have logs in NNI. They are nnimanager, dispatcher and trial. Here we will introduce them succinctly. More information please refer to [Overview](Overview.md).
+
+- **nnictl**: NNI controller is the nni command line tool that is used to started NNI.
```
Comment (QuanluZhang) [inline docs/HowToDebug.md]: `NNI controller` -> `nnictl`
`command line` -> `command-line`
`to started NNI` -> `to manage experiments (e.g., start an experiment)`

File: docs/HowToDebug.md [hunk 3]
```diff
@@ -1,4 +1,80 @@
...
 ===
 
-*Coming soon*
+## Overview
+
+There are three parts that might have logs in NNI. They are nnimanager, dispatcher and trial. Here we will introduce them succinctly. More information please refer to [Overview](Overview.md).
+
+- **nnictl**: NNI controller is the nni command line tool that is used to started NNI.
+- **nnimanager**: nnimanager is the core of NNI, whose log is important when the whole experiment fails (e.g., no webUI or training service fails)
+- **Dispatcher**: Dispatcher is the collective name of **Tuner** and **Assessor**. Logs of dispatcher are related to the tuner or assessor code.
```
Comment (QuanluZhang) [inline docs/HowToDebug.md]: `Dispatcher is the collective name of` -> `Dispatcher calls the methods of`

File: docs/HowToDebug.md [hunk 4]
```diff
@@ -1,4 +1,80 @@
...
+    - **Tuner**: Tuner is an AutoML algorithm, which generates a new configuration for the next try. A new trial will run with this configuration.
+    - **Assessor**: Assessor analyzes trial's intermediate results (e.g., periodically evaluated accuracy on test dataset) to tell whether this trial can be early stopped or not.
+- **Trial**: Trial code is the code you write to run your experiment, which is an individual attempt at applying a new configuration (e.g., a set of hyperparameter values, a specific nerual architecture).
+
+## Where is the log
+
+### nnictl
+
+All possible errors that happen when launching NNI can be found here.
```
Comment (QuanluZhang) [inline docs/HowToDebug.md]: `NNI` -> `an NNI experiment`

File: docs/HowToDebug.md [hunk 5]
```diff
@@ -1,4 +1,80 @@
...
+
+You can use `nnictl log stderr` to find error information. Or find it in `~/nni/nnictl/log/stderr`.
+
+
+### Experiment Root Directory
+Every experiment has a root folder, which is shown on the right-top corner of webUI. Or you could assemble it by replacing the `experiment_id` with your actual experiment_id in path `~/nni/experiment/experiment_id/` in case of webUI failure. `experiment_id` could be seen when you run `nnictl create ...` to create a new experiment.
```
Comment (QuanluZhang) [inline docs/HowToDebug.md]: please make it clear when the log path is configured by users.

File: docs/HowToDebug.md [hunk 6]
```diff
@@ -1,4 +1,80 @@
...
+
+Under the root path, there is a directory named `log`, where `nnimanager.log` and `dispatcher.log` are there.
```
Comment (QuanluZhang) [inline docs/HowToDebug.md]: `Under the root path` -> `Under that directory`

Comment (QuanluZhang) [reply docs/HowToDebug.md]: `are there` -> `are placed`

File: docs/HowToDebug.md [hunk 7]
```diff
@@ -1,4 +1,80 @@
...
+
+### Trial Root Directory
+
+Usually in webUI, you can click `+` in the left of every trial to expand it to see each trial's log path. Also there is another directory under experiment root directory, named `trials-local`, which stores all the trials run. Every trial has a unique id that is the name of its root directory. In this directory, a file named `stderr` records trial error and another named `trial.log` records this trial's log.
```
Comment (QuanluZhang) [inline docs/HowToDebug.md]: `trials-local`?

Comment (QuanluZhang) [reply docs/HowToDebug.md]: `trials run` -> `trials`
`that is the name of its root directory` -> `as its directory name`

File: docs/HowToDebug.md [hunk 8]
```diff
@@ -1,4 +1,80 @@
...
+
+There are different kinds of errors. However, they can be divided into three categories based on their severity. So when nni fails, check each part sequentially.
+
+Generally, if webUI is started successfully, there is a `Status` in the `Overview` tab, serving as a possible indicator of what kind of error happens. Otherwise you should check manually.
+
+### **NNI** Fails
+
+This is the most serious error. When this happens, the whole experiment fails and no trial will be run. Usually this might be related to some installation problem.
+
+When this happens, you should check `nnictl`'s error output file `stderr` and then the `nnimanager`'s log to find if there is any error.
```
Comment (QuanluZhang) [inline docs/HowToDebug.md]: `and then` ->`(i.e., nnictl log stderr) and then`

File: docs/HowToDebug.md [hunk 9]
```diff
@@ -1,4 +1,80 @@
...
+  File "/usr/lib/python3.7/site-packages/nni/__main__.py", line 164, in main
+    args.tuner_args)
+  File "/usr/lib/python3.7/site-packages/nni/__main__.py", line 81, in create_customized_class_instance
+    instance = class_constructor(**class_args)
+TypeError: __init__() missing 1 required positional arguments: 'optimize_mode'.
+```
+
+### **Trial** Fails
+
+In this situation, NNI can still run and dispatch trials.
```
Comment (QuanluZhang) [inline docs/HowToDebug.md]: `dispatch` -> `create new`

File: docs/HowToDebug.md [hunk 10]
```diff
@@ -1,4 +1,80 @@
...
+
+In this situation, NNI can still run and dispatch trials. 
+
+It means your trial code (which is run by NNI) fails. This kind of error is strongly related to your trial code. Please check trial's log to fix any possible errors shown there.
+
+A common example of this would be run the mnist example without installing tensorflow. Surely there is an Import Error (that is, not installing tensorflow but trying to import it in your trial code) and thus every trial fails.
+
+![](img/trial_error.jpg)
+
+As it shows, every trial has a log path, where you can find trial'log and stderr.
```
Comment (QuanluZhang) [inline docs/HowToDebug.md]: could you also show how to configure log level here, so that it is easy for users to debug.

Comment (Crysple) [reply docs/HowToDebug.md]: Have added in `Where it the log`

**Pull request merged**